### PR TITLE
add new ValueNode "Modulo" for modulo operation

### PR DESCRIPTION
**Pull request by rodolforg**
Suggested by woody in
https://forums.synfig.org/t/if-else-expression/12528/9

### CODE REVIEW

File: synfig-core/src/synfig/valuenodes/valuenode_modulo.cpp [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+ValueNode_Modulo::ValueNode_Modulo(const ValueBase &value):
+	LinkableValueNode(value.get_type())
+{
+	Vocab ret(get_children_vocab());
+	set_children_vocab(ret);
+	set_link("divisor",ValueNode_Const::create(int(2)));
+	set_link("scalar",ValueNode_Const::create(Real(1.0)));
+	Type& type(value.get_type());
+
+	if (type == type_angle)
```
Comment (ice0) [inline synfig-core/src/synfig/valuenodes/valuenode_modulo.cpp:77]: Why don't do something like:
```
if (!check_type(type))
{
	assert(0);
	throw std::runtime_error(get_local_name()+_(":Bad type ")+type.description.local_name);
}
set_link("link",ValueNode_Const::create(value));
```

Comment (rodolforg) [reply synfig-core/src/synfig/valuenodes/valuenode_modulo.cpp:77]: Nice.
I just based my code in another valuenode implementation (I think it was `add`) and didn't think about it :)
I'll fix it.

Comment (rodolforg) [reply synfig-core/src/synfig/valuenodes/valuenode_modulo.cpp]: Done :)

File: synfig-core/src/synfig/valuenodes/valuenode_modulo.cpp [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+/* === U S I N G =========================================================== */
+
+using namespace etl;
+using namespace synfig;
+
+/* === M A C R O S ========================================================= */
+
+/* === G L O B A L S ======================================================= */
+
+REGISTER_VALUENODE(ValueNode_Modulo, RELEASE_VERSION_1_4_0, "modulo", N_("Modulo"))
```
Comment (rodolforg) [inline synfig-core/src/synfig/valuenodes/valuenode_modulo.cpp:62]: I'm hesitant with this version detail: should we create RELEASE_VERSION_1_4_0?

Comment (ice0) [reply synfig-core/src/synfig/valuenodes/valuenode_modulo.cpp:62]: Do you mean `1_4_3`?

Comment (rodolforg) [reply synfig-core/src/synfig/valuenodes/valuenode_modulo.cpp:62]: hmmm Maybe 1_6_0?

Comment (ice0) [reply synfig-core/src/synfig/valuenodes/valuenode_modulo.cpp:62]: > hmmm Maybe 1_6_0?

Agree, it makes sense. 1.4.x is mostly bug fixes, so 1.6.0 is much better for a new feature.

**Comment by rodolforg** [comment]
@ice0 what do you think about the commit I just added?

**Review by ice0 — APPROVED** [review]

**Comment by ice0** [comment]
> @ice0 what do you think about the commit I just added?

Agree.

**Comment by ice0** [comment]
Merged. Thank you!

**Pull request merged**